### ISSUE TITLE
Building from source on a Debian ARM 64 system (Raspberry Pi 4) results in a panic

### ISSUE THREAD

**Issue by lapin-b** [timescale/promscale_extension#558]
## Context
Since there are no pre-built binaries for the database extension, and the connector requires the extension to be installed, I have decided to attempt building it from source. I know #283 is a thing to release pre-built binaries.

## Steps to reproduce
I have followed the [steps described in the INSTALL.md file](https://github.com/timescale/promscale_extension/blob/master/INSTALL.md#compile-from-source-on-ubuntu-linux) as a non-root user. The steps are:

- Install PostgreSQL 14 from the PostgreSQL repositories
- Install TimescaleDB from its APT repository
- Enable the preloading of the extension in the database's configuration file
- Restart PostgreSQL
- Verify that TimescaleDB is enabled with the presence of a "TimescaleDB Background Worker Launcher" process
- Clone this repository
- Checkout tag `0.7.0`
- Execute `install-cargo-pgx.sh`
- Initialize PGX with `cargo pgx init --pg14=/usr/lib/postgresql/14/bin/pg_config`
- Execute `make package`

## What I've got

The error printed to the terminal is:

```
    Finished release [optimized + debuginfo] target(s) in 4m 20s
 Discovering SQL entities
  Discovered 34 SQL entities: 9 schemas (1 unique), 16 functions, 1 types, 0 enums, 8 sqls, 0 ords, 0 hashes, 0 aggregates
The application panicked (crashed).
Message:  No control file found
Location: /home/l4/.cargo/git/checkouts/pgx-86cb45baab31d1a6/96ece24/pgx-utils/src/sql_entity_graph/pgx_sql.rs:137

  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ SPANTRACE ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

   0: pgx_utils::sql_entity_graph::pgx_sql::build with extension_name="promscale" versioned_so=true force_create_or_replace=true
      at pgx-utils/src/sql_entity_graph/pgx_sql.rs:81
   1: cargo_pgx::command::schema::generate_schema with pg_version=14.14 release=true test=false features=[]
      at cargo-pgx/src/command/schema.rs:145
   2: cargo_pgx::command::schema::execute
      at cargo-pgx/src/command/schema.rs:84

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
Segmentation fault
make: *** [Makefile:99: package] Error 139

```
Note: a few contextual lines have been pasted above the error.

## What I expect

A successful compilation enabling the use of the command `sudo make install`, like I've had testing on a Fedora x86 system.

## System information
- OS: Raspberry Pi OS bullseye
- Architecture: ARM64
- (edit) Compiler version: 1.65.0

**Comment by JamesGuthrie** [comment]
@lapin-b Thanks for this thorough bug report. I can reproduce the issue.

I'm pretty certain that this was a bug in pgx, which has been resolved since the 0.5.x release (in this PR: https://github.com/tcdi/pgx/pull/687). Our master branch has been upgraded to pgx 0.5.x, but we don't have a release with that change yet. With our next release, your issue should be fixed.

As you can tell from the PR, the issue is one of the underlying representation of types which are passed between binaries being partially incorrect, or otherwise relying on undefined behaviour in Rust. Until we make a new release, it's _possible_ that using a different rust compiler version would work around this issue. You would need to experiment with different compiler versions and see what you can get to work.

**Comment by lapin-b** [comment]
@JamesGuthrie Thanks for the answer!

I'll try to compile the extension using the version of the Rust compiler indicated in the Dockerfile — `1.58.1` — used to build the latest release. (The latest as of writing this comment is the 0.7.0, for anyone reading this in the future.) If that fails, I'll try compiling the master branch with the latest compiler version ─ `1.65.0`. Maybe not the best idea, but it can give some feedback about building successfully or not on ARM64.

**Comment by lapin-b** [comment]
Small update, before I forget to write it.

I have successfully compiled the extension using the Rust compiler `1.58.1`. One important note, I've made sure to execute the `install-cargo-pgx.sh` script using *the same compiler version*.

Once the extension was compiled successfully, I was able to `sudo make install` the extension, then `CREATE EXTENSION timescaledb;` and `CREATE EXTENSION promscale;` on the target database.
